packages/backend/src/vitest.setup.integration.ts

```diff
@@ -0,0 +1,239 @@
+        const getTestConnection = () => {
+            const baseUri = process.env.PGCONNECTIONURI;
+            if (!baseUri) {
+                throw new Error(
+                    'PGCONNECTIONURI environment variable is required for tests',
+                );
+            }
+
+            const connection = parse(baseUri);
+            connection.database = `${connection.database || 'lightdash'}_test`;
+
+            console.log('Using test database:', connection.database);
+            return connection;
+        };
+
+        const app = new App({
+            lightdashConfig,
+            port: 0,
+            environment: 'development',
+            knexConfig: {
+                ...knexConfig,
+                development: {
+                    ...knexConfig.development,
+                    // @ts-expect-error - TODO: fix this
+                    connection: getTestConnection(),
+                    migrations: {
+                        ...knexConfig.development.migrations,
+                        // For vitest environment, use compiled JS files instead of TS
+                        loadExtensions: ['.js'],
+                        directory: [
+                            path.join(__dirname, '../dist/database/migrations'),
+                            ...(lightdashConfig.license.licenseKey
+                                ? [
+                                      path.join(
+                                          __dirname,
+                                          '../dist/ee/database/migrations',
+                                      ),
+                                  ]
+                                : []),
+                        ],
+                    },
+                    seeds: {
+                        ...knexConfig.development.seeds,
+                        // For vitest environment, use compiled JS files instead of TS
+                        loadExtensions: ['.js'],
+                        directory: [
+                            path.join(
+                                __dirname,
+                                '../dist/database/seeds/development',
+                            ),
+                            ...(lightdashConfig.license.licenseKey
+                                ? [
+                                      path.join(
+                                          __dirname,
+                                          '../dist/ee/database/seeds/development',
+                                      ),
+                                  ]
+                                : []),
+                        ],
+                    },
+                },
+            },
+            ...enterpriseArgs,
+        });
+
+        const db = app.getDatabase();
+
+        // Run migrations to ensure database schema is up to date
+        console.log('🔧 Running database migrations...');
+        await db.migrate.latest();
+        console.log('✅ Database migrations completed');
```

Comment:
could you change all console.logs to `console.info`?

Reply:
ofc I can!

Reply:
done